File: src/main/java/com/example/demo/service/QuizService.java
package com.example.demo.service;

import java.util.List;
import java.util.Optional;
import java.util.Set;

import com.example.demo.model.exam.Category;
import com.example.demo.model.exam.Quiz;



public interface QuizService {

	public Quiz addQuiz(Quiz quiz);
	public Quiz updateQuiz(Quiz quiz);
	public Set<Quiz> getQuizzex();
	public Optional<Quiz> getQuiz(Long quizId);
	public void deleteQuiz(Long quizId);
	public List<Quiz> getQuizzesOfCat(Category category) ;
}

File: src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  public quizzes()
  {
    return this.http.get("http://localhost:8080/quiz/");
  }
  public addQuiz(Quiz:any)
  {
    return this.http.post("http://localhost:8080/quiz/",Quiz); 
  }
  public deleteQuiz(qid:number)
  {
    return this.http.delete("http://localhost:8080/quiz/"+qid);
  }
  public getQuiz(qid:number)
  {
    return this.http.get("http://localhost:8080/quiz/"+qid)
  }
  public updateQuiz(quiz:any)
  {
    return this.http.put("http://localhost:8080/quiz/",quiz);
  }

  public getQuizzesOfCat(cid:any)
  {
    return this.http.get("http://localhost:8080/quiz/category/"+cid);
  }
}
